src/Marten/Services/SessionOptions.cs

```diff
@@ -93,6 +101,23 @@ internal IConnectionLifetime Initialize(DocumentStore store, CommandRunnerMode m
             throw new DefaultTenantUsageDisabledException();
         }
 
+        var innerConnectionLifetime =GetInnerConnectionLifetime(store, mode);
+
+        var corelationId = Activity.Current is not null ? Activity.Current.RootId : Guid.NewGuid().ToString();
+        var tags = new Dictionary<string, object>
```

Comment:
Prefer `TagsList` when dealing with OTEL tags. This should also be lazy initialized only if `EventTracingConnectionLifetime` is being used.